tests/test_web_functional.py

```diff
@@ -523,7 +504,7 @@ async def handler(request):
         return web.Response()
 
     form = FormData()
-    form.add_field("name", b"123", content_transfer_encoding="base64")
+    form.add_field("name", b"123")
```

Comment:
Not sure how to fix this. Currently, if the value is bytes and content_transfer_encoding is not sent, then the code assumes that it is a file. This seems kind of dumb. Not sure how best to deal with this, given that content_transfer_encoding should never be sent.

Ironically, the spec says:
> Currently, no deployed implementations that send such bodies have been discovered.
https://datatracker.ietf.org/doc/html/rfc7578#section-4.8

Well, I just discovered one...

Reply:
On the client side, I think the only way to avoid a breaking change is to keep `content_transfer_encoding` in `add_field` for the client side but ignore it on the server side

Reply:
Are we in agreement to switch the behaviour in v4 to send files only when filename is passed (even if `filename=None`)?
This is probably the last blocker to getting 3.9.4 out.

Reply:
It looks like the example to upload a file doesn't pass filename so I expect it might be quite common

https://docs.aiohttp.org/en/stable/client_quickstart.html#post-a-multipart-encoded-file

Reply:
> It looks like the example to upload a file doesn't pass filename so I expect it might be quite common
> 
> https://docs.aiohttp.org/en/stable/client_quickstart.html#post-a-multipart-encoded-file

The first example isn't using FormData, so shouldn't be affected by this change. The second example uses FormData in order to include a filename. Which suggests to me that most cases of using FormData won't be passing a file without a filename.

Reply:
Both examples are also file objects, not bytes, further suggesting that passing bytes as a file is not a common thing in the first place.

Reply:
Actually, that does appear to just end up calling the same method on FormData in the end, so could cause an issue when passed directly in post(). The example still uses file objects though, and other parts of the documentation utilise BytesIO, so it would still seem surprising to me to get a file field when passing bytes.

Reply:
> so it would still seem surprising to me to get a file field when passing bytes.

I agree that is unexpected. It seems reasonable to have a breaking change to require `filename` instead of `bytes/bytearray/memoryview` in 4.0

Reply:
OK, reviewing again, I don't think `filename=None` would make sense, as it's the presence of a filename that makes it a FileField. The bytes version just sets filename to name, so I think we should just remove that bit of logic.

We can then add warnings, so if a use use content_transfer_encoding it tells them to use a BytesPayload directly if they want forward compatibility, and if they pass bytes it tells them to use the filename parameter or use BytesIO.